x-pack/plugins/observability/public/pages/slos/components/slos_grid/slo_grid_item.tsx

```diff
@@ -0,0 +1,211 @@
+
+export interface Props {
+  slo: SLOWithSummaryResponse;
+  rules: Array<Rule<SloRule>> | undefined;
+  historicalSummary?: HistoricalSummaryResponse[];
+  historicalSummaryLoading: boolean;
+  activeAlerts?: number;
+  loading: boolean;
+  error: boolean;
+}
+
+export const getColor = (
+  isEnabled: boolean,
+  status?: SLOWithSummaryResponse['summary']['status']
+) => {
+  if (!isEnabled) {
+    return euiLightVars.euiColorLightestShade;
+  }
+
+  switch (status) {
+    case 'DEGRADING':
+      return euiLightVars.euiColorVis7_behindText;
+    case 'VIOLATED':
+      return euiLightVars.euiColorVis9_behindText;
+    case 'HEALTHY':
+      return euiLightVars.euiColorVis0_behindText;
+    case 'NO_DATA':
+      return euiLightVars.euiColorGhost;
+    default:
+      return euiLightVars.euiColorVis0_behindText;
+  }
+};
+
+export function SLOGridItem({ slo, rules, activeAlerts, historicalSummary }: Props) {
+  const {
+    application: { navigateToUrl },
+    triggersActionsUi: { getAddRuleFlyout: AddRuleFlyout },
+  } = useKibana().services;
+
+  const [isMouseOver, setIsMouseOver] = useState(false);
+  const [isActionsPopoverOpen, setIsActionsPopoverOpen] = useState(false);
+  const [isAddRuleFlyoutOpen, setIsAddRuleFlyoutOpen] = useState(false);
+  const [isDeleteConfirmationModalOpen, setDeleteConfirmationModalOpen] = useState(false);
+
+  const { currentValue, sloTarget, sloDetailsUrl } = useSLOSummary(slo);
+  const filteredRuleTypes = useGetFilteredRuleTypes();
+  const queryClient = useQueryClient();
+  const { mutate: deleteSlo } = useDeleteSlo();
+
+  const historicalSliData = formatHistoricalData(historicalSummary, 'sli_value');
+
+  const handleSavedRule = async () => {
+    queryClient.invalidateQueries({ queryKey: sloKeys.rules(), exact: false });
+  };
+
+  const handleDeleteConfirm = () => {
+    setDeleteConfirmationModalOpen(false);
+    deleteSlo({ id: slo.id, name: slo.name });
+  };
+
+  const handleDeleteCancel = () => {
+    setDeleteConfirmationModalOpen(false);
+  };
+
+  const handleCreateRule = () => {
+    setIsActionsPopoverOpen(false);
+    setIsAddRuleFlyoutOpen(true);
+  };
+
+  return (
+    <>
+      <EuiPanel
+        onMouseOver={() => {
+          if (!isMouseOver) {
+            setIsMouseOver(true);
+          }
+        }}
+        onMouseLeave={() => {
+          if (isMouseOver) {
+            setIsMouseOver(false);
+          }
+        }}
+        paddingSize="none"
+        style={{
+          height: '200px',
+          overflow: 'hidden',
+          position: 'relative',
+        }}
+      >
+        <Chart>
+          <Settings
+            baseTheme={DARK_THEME}
+            onElementClick={([d]) => {
+              if (isMetricElementEvent(d)) {
+                navigateToUrl(sloDetailsUrl);
+              }
+            }}
+          />
+          <Metric
+            id="1"
```

Comment:
Can we give it a more dynamic id. Something like id={`${slo.id}-${slo.instanceId}`}